dwave/preprocessing/include/dwave-preprocessing/fix_variables.hpp

```diff
@@ -90,13 +91,21 @@ void fixQuboVariables(PosiformInfo &posiform_info, int num_bqm_variables,
  *      to zero when converted to the posiform.
  */
 template <class V, class B>
-std::vector<std::pair<int, int>>
+std::pair<double, std::vector<std::pair<int, int>>>
 fixQuboVariables(dimod::AdjVectorBQM<V, B> &bqm, bool strict) {
   int num_bqm_variables = bqm.num_variables();
   PosiformInfo<dimod::AdjVectorBQM<V, B>, capacity_type> posiform_info(bqm);
   std::vector<std::pair<int, int>> fixed_variables;
-  fixQuboVariables(posiform_info, num_bqm_variables, strict, fixed_variables);
-  return fixed_variables;
+  capacity_type max_flow = fixQuboVariables(posiform_info, num_bqm_variables, strict, fixed_variables);
+
+  // The max_flow is supposed to be the lower bound of the posiform which should
+  // be equal to the lower bound of the bqm. But while creating the implication
+  // network and assigning capacities to its edges we did not divide the
+  // corresponding coefficients of the posiform by 2 thus when we convert the
+  // max flow to the minimum value of the posiform we need to divide it by 2 and
+  // hence the multiplication by 0.5 here.
```

Comment:
Might be nice to link the paper here. Also could remove "and hence the multiplication by 0.5 here".